mlflow/pyfunc/__init__.py

```diff
@@ -201,21 +202,65 @@ def spark_udf(spark, path, run_id=None, result_type="double"):
     Parameters passed to the UDF are forwarded to the model as a DataFrame where the names are
     ordinals (0, 1, ...).
 
+    The predictions are filtered to contain only the columns that can be represented as the
+    ``result_type``. If the ``result_type`` is string or array of strings, all predictions are
+    converted to string. If the result type is not an array type, the left most column with
+    matching type will be returned.
+
+    Note: Exception is raised if there is no matching type in the result.
+
+    Note: It is recommended to only use one of LongType, DoubleType, StringType or ArrayType of
+    one of these types as the result_type.
+
     >>> predict = mlflow.pyfunc.spark_udf(spark, "/my/local/model")
     >>> df.withColumn("prediction", predict("name", "age")).show()
 
     :param spark: A SparkSession object.
     :param path: A path containing a :py:mod:`mlflow.pyfunc` model.
     :param run_id: ID of the run that produced this model. If provided, ``run_id`` is used to
                    retrieve the model logged with MLflow.
+    :param result_type: the return type of the user-defined function. The value can be either a
+                        :class:`pyspark.sql.types.DataType` object or a DDL-formatted type string.
+                        Only a primitive type or an array of primitive types are allowed.
+                        The following classes of result type are supported:
+
+                        - "int", "long" or instanceof IntegralType: The leftmost integer result is
+                          returned or exception is raised if there is none. No type conversion is
+                          performed.
+                        - ArrayType(instanceof IntegralType): Return all integer columns. Exception
+                          is raised if there are none.
+                        - "float" or "double" or instanceof FractionalType: The leftmost numeric
+                          result is returned or exception is raised if there is none.
+                        - ArrayType(instanceof FractionalType): Return all numeric columns.
+                          Exception is raised if there are none. No type conversion is performed.
+                        - "string" or StringType: Result is the leftmost column converted to string.
+                        - Array[StringType]: Return all columns converted to string.
+
+                        NOTE: Conversion may fail if e.g. element type is set to IntegerType and the
```

Comment:
Can we specify that the conversion failure comes from PySpark (PyArrow)? How do you feel about using the following text instead?:

```
NOTE: The model output type must be compatible with the specified PySpark ``result_type``. For example, if ``result_type`` is set to IntegerType and the model returns longs instead, PySpark type conversion will fail.
```